share/picongpu/tests/compileParticlePusher/include/picongpu/param/species.param

```diff
@@ -17,54 +17,74 @@
  * If not, see <http://www.gnu.org/licenses/>.
  */
 
+/** @file
+ *
+ * Forward declarations for speciesDefinition.param in case one wants to use
+ * the same particle shape, interpolation, current solver and particle pusher
+ * for all particle species.
+ */
+
 #pragma once
 
 #include "picongpu/particles/shapes.hpp"
 #include "picongpu/algorithms/FieldToParticleInterpolationNative.hpp"
 #include "picongpu/algorithms/FieldToParticleInterpolation.hpp"
 #include "picongpu/algorithms/AssignedTrilinearInterpolation.hpp"
-
 #include "picongpu/particles/flylite/NonLTE.def"
 #include "picongpu/fields/currentDeposition/Solver.def"
 
 
 namespace picongpu
 {
-/*---------------------------- generic solver---------------------------------*/
 
-/*! Particle Shape definitions -------------------------------------------------
- *  - particles::shapes::CIC : 1st order
- *  - particles::shapes::TSC : 2nd order
- *  - particles::shapes::PCS : 3rd order
- *  - particles::shapes::P4S : 4th order
- *
- *  example: using UsedParticleShape = particles::shapes::CIC;
+/** Particle Shape definitions
+ *  Particle shape naming is according to
+ *    Hockney R, Eastwood JW. Computer simulation using particles. IOP Pubslishing Ltd (1988). p. 144.
+ *  That is, the name of a particle shape is defined by the visual form of its cloud shape S,
+ *  which describes a particle's charge density distribution.
+ *  However, the charge assigned to a mesh point due to a distribution of overlapping finite-sized
+ *  particles is obtained by evaluating and summing the assignment function W for all particles.
+ *  The assignment function is obtained by convolving the cloud shape with the top-hat function.
+ *  PIConGPU only implements the assignment functions.
```

Comment:
I am not sure this information is relevant at this point. I would imagine a user cares about how to set the desired properties, not how it's implemented internally

Reply:
I would also move such information to the documentation instead of having it here.
Should we also shorten this section and move it to the documentation and link the documentation here:

```
...

**WARNING** the shape naming has changed and is not consistent too PIConGPU versions before 0.6.0. see readthedocslink

  - particles::shapes::CIC : 1st order, assignment function is a piecewise linear spline
  - particles::shapes::TSC : 2nd order, assignment function is a piecewise quadratic spline
  - particles::shapes::PQS : 3rd order, assignment function is a piecewise cubic spline
  - particles::shapes::PCS : 4th order, assignment function is a piecewise quartic spline
```

Reply:
I moved most of the text to the general file documentation, where I think it belongs and is visible at readthedocs. I also think we can leave the explaining sentence on PIConGPU imlementing only assignment functions at this general file documentation level, as it will help users/developers understanding what PIConGPU does. I agree, it is superfluous as a comment to `using UsedParticleShape`.